compiler/rustc_trait_selection/src/traits/select/confirmation.rs

```diff
@@ -306,7 +306,7 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
                 .substs
                 .const_at(i)
                 .try_eval_bool(self.tcx(), obligation.param_env)
-                .unwrap()
+                .unwrap_or(true)
```

Comment:
heh, That's smart. Should probably document this behaviour in the tracking issue so it can get revisited before stabilization.

Reply:
I don't believe this is a stability issue (even if the feature was on track for stabilization, which it isn't). The `unwrap_or(true)` branch is only reached if the const parameter is of the wrong type. This, in itself, is a type error, which rustc reports separately. However, since rustc nonetheless still tries to confirm this candidate, the compiler would ICE had I only written `.unwrap()`. So, what to do instead? I could:
- bail out of the whole confirmation process with `Err(Unimplemented)`
  - this would always result in a second, "not transmutable" error message being shown to the user
- bail out of the whole confirmation process with `Ok(ImplSourceBuiltinData { nested: vec![] }),`
  - this would never result in a "not transmutable" error message being shown, even if there's no way the types are safely transmutable even if that mis-provided safety check turned off
 - conservatively assume that the user wants the safety check turned off
   - this still results in an error message if the types cannot possibly be transmutable
   - but does not display an error message if the types could be transmutable under the right conditions
     - ...and those "right conditions" will become well-defined as soon as the user fixes the other error and passes in the const parameter correctly

In all possibilities, at least one error message is always displayed: the error that a const parameter of the wrong type has been passed in. So, whether or not the transmutability system *also* produces its own error isn't a stability issue, it's just an error message ergonomics issue.